packages/block-library/src/details/edit.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	RichText,
+	useBlockProps,
+	useInnerBlocksProps,
+	BlockControls,
+} from '@wordpress/block-editor';
+import { ToolbarButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import HeadingLevelDropdown from './heading-level-dropdown';
+
+const DETAILS = [
+	[
+		'core/paragraph',
+		{
+			placeholder: __(
+				'Add text or blocks that will display when the details block is opened.'
+			),
+		},
+	],
+];
+
+function DetailsBlock( { attributes, setAttributes } ) {
+	const { level, summary, showContent } = attributes;
+	const tagName = 'h' + level;
```

Comment:
Is there a case where headings wouldn't make semantic sense? That is, should we also allow body text (`<span />`) or something?

Reply:
@joedolson what do you think? Could the content in `<summary>` use other tags than headings?
It is valid HTML but do we want to allow it?

Reply:
Pretty much any inline tag can be used in `summary`, though some of the allowed tags would be a bit strange, like `audio`. 

Restricting the content to a heading is what makes the most sense to me. Even if other tags are allowed, they could be attached within the heading (e.g., a heading containing `span`, `strong`, `mark`, etc.)

I can imagine cases where the heading is not super helpful (e.g., if somebody uses the block to reveal single images with no alt text, then there's effectively no contained content, and the heading structure has no meaning); but I think that's more a misuse of the UI than a problem with the structure of the UI.

Reply:
I think that nesting the rich text inside a heading inside the summary would be unnecessary complex, especially if we want to be able to select the HTML tag for both the heading and the inner element.

Reply:
There's a big problem here: a `summary` element breaks the semantics of headings.

https://daverupert.com/2019/12/why-details-is-not-an-accordion/

Reply:
I only tested with VoiceOver which did include the heading inside the summary when navigating via headings.

Reply:
Though I would rather have the block with a plain `<summary>` without inner element, than not have it available at all.

Reply:
@ZebulanStanphill Note that article is 3 years old; given the rate of change in the accessibility API stack, it shouldn't be considered a strong reference. There are mixed opinions about whether headings are good or bad inside a `summary` element, so it's a decision that we would need to make. In my opinion, we should include a heading inside the summary.

Here is some testing data: https://a11ysupport.io/tests/tech__html__details-summary#assertion-html-h1-6_elements-convey_role_and_name-

Note that 'fail' means "heading semantics were ignored by the screen reader", which is not the same as "this is not accessible". The open question is whether or not a `summary` should have both control and heading semantics, and not all API trees/AT agrees on this point. However, it does mean that the relationship between what a simple testing tool (like HeadingsMap) will tell you about the heading hierarchy on your page and the real experience that an AT user will have is broken. 

One argument in favor of allowing element choice inside the summary is that it allows a user to make their own decision about that.

Reply:
@joedolson Good point.

@carolinan I agree that a `<details>` block is desirable regardless of whether or not headings are ultimately allowed in its `<summary>`.